src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/FinalLocalVariableCheck.java

```diff
@@ -363,16 +363,33 @@ private void updateUninitializedVariables(Deque<DetailAST> prevScopeUnitializedV
     }
 
     /**
-     * If token is LITERAL_TRY, LITERAL_CATCH, LITERAL_FINALLY, or LITERAL_ELSE, then do not
-     * update the uninitialized variables.
+     * If token is LITERAL_IF and there is an {@code else} following or token is CASE_GROUP and
+     * there is another {@code case} following, then update the uninitialized variables.
      * @param ast token to be checked
      * @return true if should be updated, else false
      */
     private static boolean shouldUpdateUninitializedVariables(DetailAST ast) {
-        return ast.getType() != TokenTypes.LITERAL_TRY
-                && ast.getType() != TokenTypes.LITERAL_CATCH
-                && ast.getType() != TokenTypes.LITERAL_FINALLY
```

Comment:
> Regrading the test. I don't understand clearly. 
> The original code means that if the ast type is not one of the first four (try, catch, finally, else), then it should update. And current code means that if the ast type is not one of try, catch, finally, method def, lambda ... then it should be update.

Before your change, we had 100% branch and code coverage. So we should have had some test that at least looked at true/false conditions of that method for `catch`/`finally`. Since you changed the method to basically allow more tokens, I would expect this to have some type of change on tests. Either violation turning no violation or vice versa. Since there are no changes, it makes me think we are missing more regression cases. What I was asking for was more examples where you change had an impact in violations to show your change is valid and won't lead to more false positives.
I didn't notice the change in conditions when first looking.
Regression will help show if there are any differences besides issue's example.